be/src/agent/heartbeat_server.cpp

```diff
@@ -95,8 +95,15 @@ Status HeartbeatServer::_heartbeat(const TMasterInfo& master_info) {
             return Status::InternalError("fail to set cluster id.");
         } else {
             _master_info->cluster_id = master_info.cluster_id;
```

Comment:
This should be set after if clause, because return error when `if clause` is true.